akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -89,18 +89,20 @@ private[akka] object RemoteActorRefProvider {
     import EndpointManager.Send
 
     override def !(message: Any)(implicit sender: ActorRef): Unit = message match {
-      case Send(m, senderOption, _, seqOpt) ⇒
+      case Send(m, senderOption, recipient, seqOpt) ⇒
         // else ignore: it is a reliably delivered message that might be retried later, and it has not yet deserved
         // the dead letter status
-        if (seqOpt.isEmpty) super.!(m)(senderOption.orNull)
+        if (seqOpt.isEmpty) super.!(DeadLetter(m, senderOption.getOrElse(_provider.deadLetters), recipient))
```

Comment:
The reason I changed this is that deadletters from remoting didn't contain the `recipient`, which can be very useful when debugging.